### PR TITLE
Dashboard link Visible and Clickable test

### PR DESCRIPTION
**Pull request by Jenny-ai**
https://trello.com/c/VrtoJcAS/30-dashboard-lower-dropdown-items-visible
https://trello.com/c/1ecQOlpK/73-dashboard-lower-dropdown-items-clickable
https://trello.com/c/ZR3wbRuh/41-dashboard-down-arrow-visible-and-clickable
https://trello.com/c/3lzbYl3v/74-dashboard-link-visible-and-clickable

added DashboardLowerDropdownPage, Locators for this Page, 1 test for Dashboard link Visible and Clickable

### CODE REVIEW

File: pages/BasePage.py [hunk 1]
```diff
@@ -42,3 +43,14 @@ def get_element_text(self, locator):
...
+    def hover_over_element(self, locator: tuple):
+        action = ActionChains(self.driver)
+        element = self.driver.find_element(locator[0], locator[1])
+        action.move_to_element(element).perform()
+
+    def hover_and_click_element(self, locator1: tuple, locator2: tuple):
+        action = ActionChains(self.driver)
+        elementToHoverOver = self.driver.find_element(locator1[0], locator1[1])
+        elementToClick = self.driver.find_element(locator2[0], locator2[1])
+        action.move_to_element(elementToHoverOver).click(elementToClick).perform()
```
Comment (MichaelMicrof) [inline pages/BasePage.py:56]: add please empty line at the end

File: pages/DashboardLowerDropdownPage.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from pages.BasePage import BasePage
+
+
+class DashboardLowerDropdownPage(BasePage):
+
+    def __init__(self, driver):
+        super().__init__(driver)
```
Comment (MichaelMicrof) [inline pages/DashboardLowerDropdownPage.py:7]: add please empty line at the end

File: tests/test_dashboard_lower_dropdown_page.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        driver = DashboardLowerDropdownPage(self.driver)
+        driver.hover_over_element(DashboardLowerDropdownPageLocators.DASHBOARD_LINK)
+        assert driver.is_visible(DashboardLowerDropdownPageLocators.MENU_SELECTOR)
+        assert driver.is_clickable(DashboardLowerDropdownPageLocators.MENU_SELECTOR)
+
+    def test_dashboard_lower_dropdown_items_clickable(self):
+        driver = DashboardLowerDropdownPage(self.driver)
+        driver.hover_and_click_element(DashboardLowerDropdownPageLocators.DASHBOARD_LINK,
+                                            DashboardLowerDropdownPageLocators.MENU_SELECTOR)
+        assert driver.is_clickable(DashboardLowerDropdownPageLocators.NEW_ITEM)
```
Comment (MichaelMicrof) [inline tests/test_dashboard_lower_dropdown_page.py:21]: add please empty line at the end

File: tests/locatorsDashboardLowerDropdown.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+from selenium.webdriver.common.by import By
+
```
Comment (MichaelMicrof) [inline tests/locatorsDashboardLowerDropdown.py:2]: delete this line, please

File: pages/DashboardLowerDropdownPage.py (hunk 2 above)
Comment (Jenny-ai) [reply pages/DashboardLowerDropdownPage.py:7]: done

File: tests/locatorsDashboardLowerDropdown.py (hunk 4 above)
Comment (Jenny-ai) [reply tests/locatorsDashboardLowerDropdown.py:2]: done

File: tests/test_dashboard_lower_dropdown_page.py (hunk 3 above)
Comment (Jenny-ai) [reply tests/test_dashboard_lower_dropdown_page.py:21]: done

File: pages/BasePage.py (hunk 1 above)
Comment (Jenny-ai) [reply pages/BasePage.py]: done

File: tests/test_dashboard_lower_dropdown_page.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        assert driver.is_clickable(DashboardLowerDropdownPageLocators.MANAGE_JENKINS)
+        assert driver.is_clickable(DashboardLowerDropdownPageLocators.MY_VIEWS)
+        assert driver.is_clickable(DashboardLowerDropdownPageLocators.LOCKABLE_RESOURCES)
+        assert driver.is_clickable(DashboardLowerDropdownPageLocators.NEW_VIEW)
+
+    def test_dashboard_lower_dropdown_items_visible(self):
+        driver = DashboardLowerDropdownPage(self.driver)
+        driver.hover_element1_and_click_element2(DashboardLowerDropdownPageLocators.DASHBOARD_LINK,
+                                                 DashboardLowerDropdownPageLocators.MENU_SELECTOR)
+        assert driver.is_visible(DashboardLowerDropdownPageLocators.NEW_ITEM)
```
Comment (Kirikami) [inline tests/test_dashboard_lower_dropdown_page.py:33]: Create new list with all elements that youre validating cause youre using same elements in 2 test method

File: tests/locatorsDashboardLowerDropdown.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+class DashboardLowerDropdownPageLocators:
+
+    DASHBOARD_LINK = (By.XPATH, "//a[text()='Dashboard']")
+    MENU_SELECTOR = (By.ID, 'menuSelector')
+    RIGHT_ARROW_SELECTOR = (By.CLASS_NAME, 'children')
+    RIGHT_ARROW_SELECTOR_ALL = (By.XPATH, '//a[@href="/view/all/"]')
+    RIGHT_ARROW_SELECTOR_ALL_VISIBLE = (By.ID, 'breadcrumb-menu')
```
Comment (Kirikami) [inline tests/locatorsDashboardLowerDropdown.py:9]: Please rename locator

File: tests/test_dashboard_lower_dropdown_page.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        driver = DashboardLowerDropdownPage(self.driver)
+        driver.hover_over_element(DashboardLowerDropdownPageLocators.DASHBOARD_LINK)
+        assert driver.is_visible(DashboardLowerDropdownPageLocators.MENU_SELECTOR)
+        assert driver.is_clickable(DashboardLowerDropdownPageLocators.MENU_SELECTOR)
+
+    def test_dashboard_lower_dropdown_items_clickable(self):
+        driver = DashboardLowerDropdownPage(self.driver)
+        driver.hover_element1_and_click_element2(DashboardLowerDropdownPageLocators.DASHBOARD_LINK,
+                                                 DashboardLowerDropdownPageLocators.MENU_SELECTOR)
+        assert driver.is_clickable(DashboardLowerDropdownPageLocators.NEW_ITEM)
```
[the remark this answers is not in the document]
Comment (Jenny-ai) [reply tests/test_dashboard_lower_dropdown_page.py]: added parametrize. It does not close and open browser.

File: tests/locatorsDashboardLowerDropdown.py (hunk 6 above)
Comment (Jenny-ai) [reply tests/locatorsDashboardLowerDropdown.py]: removed un necessary locators

File: tests/test_dashboard_lower_dropdown_page.py (hunk 5 above)
Comment (Jenny-ai) [reply tests/test_dashboard_lower_dropdown_page.py]: added parametrize

**Review by irinabarnes — APPROVED** [review]

**Pull request merged**